core/ordersync/ordersync.go

```diff
@@ -291,7 +290,7 @@ func (s *Service) GetOrders(ctx context.Context, minPeers int) error {
 
 		m := &sync.RWMutex{}
 		wg := &sync.WaitGroup{}
-		semaphore := make(chan struct{}, maxRequestPeersInParallel)
+		semaphore := make(chan struct{}, minPeers)
 		currentNeighbors := s.node.Neighbors()
 		shufflePeers(currentNeighbors)
 		innerCtx, cancel := context.WithCancel(ctx)
```

Comment:
`defer cancel()` won't run until `GetOrders` returns, but I think we want to cancel `innerCtx` after each iteration of `for len(successfullySyncedPeers) < minPeers`. Otherwise we will keep creating new contexts without canceling them.